### DISCUSSION TITLE
Setting of an adiabatic wall surrounding a fuel inflow

### DISCUSSION THREAD

**Discussion by hong27** [AMReX-Combustion/PeleLMeX#390] [Q&A] [answered]
Hi all, 

I am currently working on setting up a case with pipe walls surrounding the fuel inflow. A schematic shown as below.
<img width="506" alt="image" src="https://github.com/AMReX-Combustion/PeleLMeX/assets/49507355/3b8510cb-f4a7-48c5-a041-444d6bb627ab">
For an isothermal wall, this can be achieved using the zero_visc function as demonstrated in the DiffBunsen2D case ($PELE_HOME/Exec/Production/DiffBunsen2D) via the zero_visc function. 
```
// the following extracted from $PELE_HOME/Exec/Production/DiffBunsen2D/pelelmex_prob.H

zero_visc(
  int i,
  int j,
  int k,
  amrex::Array4<amrex::Real> const& beta,
  amrex::GeometryData const& geomdata,
  amrex::Box const& domainBox,
  const int dir,
  const int beta_comp,
  const int nComp)
{
  amrex::ignore_unused(
    i, j, k, beta, geomdata, domainBox, dir, beta_comp, nComp);

  // We treat species when beta_comp == 0 and nComp == NUM_SPECIES
  // otherwise this routine could be called for other face diffusivity (Temp,
  // velocity, ...)
  if ((beta_comp == 0) && (nComp == NUM_SPECIES)) {
    if (dir == 1 && j <= domainBox.smallEnd(dir)) {
      // In the zone 2, we have an obstacle (fuel pipe). Thus, the flow is cut.
      const amrex::Real* prob_lo = geomdata.ProbLo();
      const amrex::Real* dx = geomdata.CellSize();
      const amrex::Real x = prob_lo[0] + (i + 0.5) * dx[0];

      if ((x > 0.0011) and (x < 0.0012)) // Hard coded size of the burner lip
                                         // (we can't use void get_zone as
                                         // Prob_parm is not called in this
                                         // function
      {
        for (int n = beta_comp; n < beta_comp + nComp; n++) {
          beta(i, j, k, n) = 0.0;
        }
      }
    }
  }
}
```
However, I intend to set up this wall as an adiabatic one. I am currently considering to modify the zero_visc function with a few lines to accomplish this. Specifically by trying to setting the thermal conductivity to zero, along with the zero diffusivity of species. 
```
zero_visc(
  int i,
  int j,
  int k,
  amrex::Array4<amrex::Real> const& beta,
  amrex::GeometryData const& geomdata,
  amrex::Box const& domainBox,
  const int dir,
  const int beta_comp,
  const int nComp)
{
  amrex::ignore_unused(
    i, j, k, beta, geomdata, domainBox, dir, beta_comp, nComp);

  // We treat species when beta_comp == 0 and nComp == NUM_SPECIES
  // otherwise this routine could be called for other face diffusivity (Temp,
  // velocity, ...)
  if ((beta_comp == 0) && (nComp == NUM_SPECIES)) {
    if (dir == 1 && j <= domainBox.smallEnd(dir)) {
      // In the zone 2, we have an obstacle (fuel pipe). Thus, the flow is cut.
      const amrex::Real* prob_lo = geomdata.ProbLo();
      const amrex::Real* dx = geomdata.CellSize();
      const amrex::Real x = prob_lo[0] + (i + 0.5) * dx[0];

      if ((x > 0.0011) and (x < 0.0012)) // Hard coded size of the burner lip
                                         // (we can't use void get_zone as
                                         // Prob_parm is not called in this
                                         // function
      {
        for (int n = beta_comp; n < beta_comp + nComp; n++) {
          beta(i, j, k, n) = 0.0;
        }
      }
    }
  }


// the following lines added to include zero thermal conductivity  
   else if((beta_comp == NUM_SPECIES) &&(nComp == 1)){

    if (dir == 1 && j <= domainBox.smallEnd(dir)) {

      const amrex::Real* prob_lo = geomdata.ProbLo();
      const amrex::Real* dx = geomdata.CellSize();
      const amrex::Real x = prob_lo[0] + (i + 0.5) * dx[0];

      if ((x > 0.0011) and (x < 0.0012)){
        beta(i,j,k,0) = 0.0;
      }
    }
  }

}
```
Could you please advise if this method correctly sets up the wall surrounding the fuel inflow as an adiabatic wall? Do we need other modifications on the source code to set up the adiabatic wall condition via the zero_visc function?

Thanks.
Jianhong

**Comment by esclapez** [comment, marked as answer]
Hi Jianhong,

Assuming your inflow with pipe wall is on the low-y direction (as in the DiffBunsen2D case), this would work indeed.

To clarify why, the `zero_visc` function is called in the `getDiffusivity` function which computes face-centered transport coefficients by averaging cell-centered coefficients. The cell-centered data MultiFab has `NUM_SPECIES+1` components for species and temperature, but the face-centered one only has `nComp` components. The `beta_comp` refers to the index in the cell-centered container. So:
 - when called with `beta_comp == 0` and `nComp == NUM_SPECIES`, `getDiffusivity` (and thus `zero_visc`) is called to handle species diffusivities. 
 - when called with `beta_comp == NUM_SPECIES` and `nComp == 1`, `getDiffusivity` (and thus `zero_visc`) is called to handle
thermal diffusivity. Because the 'beta' in `zero_visc` is the face-centered data, zeroing the `beta(i,j,k,0)` is the right component to access.

This logic is definitely not user-friendly, but you seem to have gotten it right !

Cheers

**Comment by hong27** [comment]
Thanks @esclapez for the clarification. Much appreciated!
